### ISSUE TITLE
[MongoDB] Error: Authentication failed

### ISSUE THREAD

**Issue by mmlanger** [microsoft/vscode-cosmosdb#1081]
I do have problems with the vscode extension connecting to a locally running mongodb server.
The connection-string is of the type:
mongodb://myrootuser:mypassword@127.0.0.1:27017

Everything works with the 0.8.0 release and prior, but beginning with 0.9.0 and later I was not able to use the extension with my mongodb instance.

Working state (i.e. with 0.8.0):
![image](https://user-images.githubusercontent.com/14871538/55159579-a5644400-5161-11e9-90a9-4d765c3d3579.png)

Non-working state (i.e. with 0.10.0):
![image](https://user-images.githubusercontent.com/14871538/55159632-d3498880-5161-11e9-8225-c66c3283bcad.png)

The "OUTPUT" tab of the Mongo Language Server says:
```console
Debugger listening on ws://127.0.0.1:6005/92e3661d-9f2b-49b9-bb16-36257074eca9
For help, see: https://nodejs.org/en/docs/inspector
[Error - 1:59:45 PM] (node:16492) UnhandledPromiseRejectionWarning: MongoError: Authentication failed.
    at Function.MongoError.create (/home/mlanger/.vscode/extensions/ms-azuretools.vscode-cosmosdb-0.10.0/dist/node_modules/mongodb-core/lib/error.js:31:11)
    at /home/mlanger/.vscode/extensions/ms-azuretools.vscode-cosmosdb-0.10.0/dist/node_modules/mongodb-core/lib/connection/pool.js:497:72
    at authenticateStragglers (/home/mlanger/.vscode/extensions/ms-azuretools.vscode-cosmosdb-0.10.0/dist/node_modules/mongodb-core/lib/connection/pool.js:443:16)
    at Connection.Pool.socketCount [as messageHandler] (/home/mlanger/.vscode/extensions/ms-azuretools.vscode-cosmosdb-0.10.0/dist/node_modules/mongodb-core/lib/connection/pool.js:477:5)
    at Socket.dataHandler (/home/mlanger/.vscode/extensions/ms-azuretools.vscode-cosmosdb-0.10.0/dist/node_modules/mongodb-core/lib/connection/connection.js:333:22)
    at Socket.emit (events.js:182:13)
    at addChunk (_stream_readable.js:279:12)
    at readableAddChunk (_stream_readable.js:264:11)
    at Socket.Readable.push (_stream_readable.js:219:10)
    at TCP.onread (net.js:636:20)
[Error - 1:59:45 PM] (node:16492) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 1)
[Error - 1:59:45 PM] (node:16492) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
```

Please let me know If you need any further information!

**Comment by PrashanthCorp** [comment]
Hi @mmlanger , 
I tried repro'ing the issue - it looks like the issue reproduces even in 0.8.0. Stepping through the code, the auth failure is coming from Mongo itself. Looking around, it looks like this could be a root cause: 
https://jira.mongodb.org/browse/SERVER-17459

**Comment by psined1** [comment]
getting the same error with 0.10.1. Any update?

**Comment by PrashanthCorp** [comment]
I figure it could be down to the version of the mongodb package shipped. 0.8.0 has an older version of `mongodb` (2.2.25) where this works, while the newer version (2.2.36) is throwing me an authentication failed error. We have made changes to how we handle connection strings, but my repro (with debugging) didn't show me any difference for this case. I'll look at it again, and let you know of any workaround.

**Comment by psined1** [comment]
Our MongoDB version is 3.2.13

**Comment by mmlanger** [comment]
Mine is MongoDB server version: 4.0.6.

**Comment by PrashanthCorp** [comment]
Hi, I've tried reproducing this again, but to no avail. 

Mongo v3.4, Os Windows 10, extension at commit 351180b1e8016853e82306d1b7d3083fe333e123

Here's what I did: 
- Create an admin user, with a password. 
- Stop and restart mongod for the above change to take effect
- Run `mongo` then `connect("mongodb://admin:password@127.0.0.1:27017")` . I can see collections in the shell
- Detach local `127.0.0.1:27017` from the extension, and attach `mongodb://admin:password@127.0.0.1:27017` . I can see the database and accounts. 


Is there something w.r.t role based or collection based access I need to be aware of?
